src/EventStore/DbalEventStore.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Landingi\EventStoreBundle\EventStore;
+
+use Doctrine\DBAL\Connection;
+use Landingi\EventStoreBundle\Event;
+use Landingi\EventStoreBundle\EventStore;
+
+final class DbalEventStore implements EventStore
+{
+    private Connection $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     * @throws \JsonException
+     */
+    public function store(Event $event): void
+    {
+        $this->connection->insert('events_store', [
+            'account_uuid' => $event->getAccountUuid()->jsonSerialize(),
+            'aggregate_name' => $event->getAggregateName()->jsonSerialize(),
+            'aggregate_uuid' => $event->getAggregateUuid()->jsonSerialize(),
+            'created_at' => $event->getCreatedAt()->jsonSerialize(),
+            'event_name' => $event->getName()->jsonSerialize(),
+            'event_data' => json_encode($event->getEventData()->jsonSerialize(), JSON_THROW_ON_ERROR),
```

Comment:
Przy `json_encode` z tego co pamiętam `jsonSerialize` sam się odpala na obiekcie :)

Reply:
No DBAL chyba sobie tego nie zserializuje sam XD

Ale mogę się mylić. Jak wspominałem na razie implementacje przekopiowałem XD - z drobnymi poprawkami.

Reply:
Tutaj przekazany jest jeszcze dodatkowy parametr, który skutkuje tym, ze w trakcie wystąpienia błędu przy serializacji rzucany jest wyjatek JsonException.  Mamy wtedy fajne info o tym, że posypala się serializacja a nie insert sam do bazy sam w sobie.

https://www.php.net/manual/en/class.jsonexception.php

Reply:
Spoko wiem o tym. Tylko interface `JsonSerialize` jest automatem odpalany przy uzyciu funkcji `json_encode`

Reply:
Tak tak, wczytalem sie bardziej, bo Zbyszek mi tu na zywo w biurze przed chwila zadal pytanie troche inaczej i odpisalem to co powiedzial na zywo :D Mea culpa